### ISSUE TITLE
Method with custom class parameters are not generated in osx

### ISSUE THREAD

**Issue by hs3366677** [mono/CppSharp#1195]
##### Brief Description

OS:  OS X 
CppSharp Version: 0.8.23

0.8.23 works well for me on Windows. But when I use my compilation on mac osx, member methods involving custom class as passing parameters are not generated. Neither is singleton method.
For example,
class Foo1 {
}
class Foo2{
public:
static Foo instance(); 
void method1(int x);
void method2(foo1 x);
}
class Foo1 and Foo2 are both correctly generated. But instance() and method2(foo1 x) do not appear in the generated file.

##### Used settings

Target: MSVC/GCC/Clang

Other settings

**Comment by tritao** [comment]
Do you have symbols checking enabled? 
It's the only thing that comes to mind that could explain this behavior.

**Comment by tritao** [comment]
Did you figure it out meanwhile?

**Comment by tritao** [comment]
Any update?

**Comment by hs3366677** [comment]
no, it still didn't work out for me with symbols checking enabled. 
I commented out 
if (output.TranslationUnit.Module != Options.SystemModule)
in Driver.cs and it worked. Seems that the custom class was reckoned as system module ?
